src/applications/personalization/profile-2/tests/components/personal-information/PersonalInformation.update-address.unit.spec.jsx

```diff
@@ -0,0 +1,275 @@
+async function testQuickFailure(addressName) {
+  server.use(...mocks.transactionFailed);
+
+  updateAddress(addressName);
+
+  // expect an error to be shown
+  const alert = await view.findByTestId('edit-error-alert');
+  expect(alert).to.have.descendant('div.usa-alert-error');
+  expect(alert).to.contain.text(
+    `We’re sorry. We couldn’t update your ${addressName.toLowerCase()}. Please try again.`,
+  );
+
+  // make sure that edit mode is not automatically exited
+  await wait(75);
+  expect(view.getByTestId('edit-error-alert')).to.exist;
+  const editButton = getEditButton();
+  expect(editButton).to.not.exist;
+}
+
+// When the update fails but not until after the Edit View has exited and the
+// user returned to the read-only view
+async function testSlowFailure(addressName) {
+  server.use(...mocks.transactionPending);
+
+  const { cityInput } = updateAddress(addressName);
+
+  // wait for the edit mode to exit
+  await waitForElementToBeRemoved(cityInput);
+
+  // check that the "we're working on saving your..." message appears
+  const updatingMessage = await view.findByText(
+    new RegExp(`We’re working on saving your new ${addressName}.`, 'i'),
+  );
+  expect(updatingMessage).to.exist;
+
+  server.use(...mocks.transactionFailed);
+
+  await waitForElementToBeRemoved(updatingMessage);
+
+  // make sure the error message appears
+  expect(
+    view.getByText(
+      /We couldn’t save your recent .* update. Please try again later/i,
+    ),
+  ).to.exist;
+
+  // and the add/edit button should be back
+  expect(getEditButton(addressName)).to.exist;
+}
+
+describe('Updating', () => {
+  before(() => {
+    // before we can use msw, we need to make sure that global.fetch has been
+    // restored and is no longer a sinon stub.
+    resetFetch();
+    server = setupServer(...mocks.editAddressSuccess);
+    server.listen();
+  });
+  beforeEach(() => {
+    window.VetsGov = { pollTimeout: 1 };
+    const initialState = createBasicInitialState();
+
+    view = renderWithProfileReducers(ui, {
+      initialState,
+    });
+  });
+  afterEach(() => {
+    server.resetHandlers();
+  });
+  after(() => {
+    server.close();
+  });
+
+  // the list of address fields that we need to test
+  const addresses = [
+    FIELD_NAMES.MAILING_ADDRESS,
+    FIELD_NAMES.RESIDENTIAL_ADDRESS,
+  ];
+
+  addresses.forEach(address => {
+    const addressName = FIELD_TITLES[address];
+    describe(`${addressName} when the entered address passes address validation`, () => {
+      it('should handle a transaction that succeeds quickly', async () => {
+        await testQuickSuccess(addressName);
+      });
+      it('should handle a transaction that does not succeed until after the edit view exits', async () => {
+        await testSlowSuccess(addressName);
+      });
+      // TODO: this test does not pass. If the address validation API fails, we exit edit mode when we should actually keep it open. So we should fix the edit address flow to make this pass.
```

Comment:
Good catch, do we have a ticket for this that also mentions to un-skip this test?

Reply:
Good call. I will add that before I merge this

Reply:
I actually fixed this. I was setting up my API mock incorrectly :-|